packages/jellyfish-api-core/__tests__/category/loan/updateLoanScheme.test.ts

```diff
@@ -0,0 +1,185 @@
+    expect(loanSchemeId.length).toStrictEqual(64)
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloanschemes')
+    const result = data.filter((d: { id: string }) => d.id === 'scheme1')
+    expect(result.length).toStrictEqual(1)
+    expect(result[0]).toStrictEqual(
+      {
+        default: false,
+        id: 'scheme1',
+        interestrate: 3.5,
+        mincolratio: 300
+      }
+    )
+  })
+
+  it('should not updateLoanScheme if minColRatio is less than 100', async () => {
+    const promise = testing.rpc.loan.updateLoanScheme(99, new BigNumber(3.5), { id: 'scheme1' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nminimum collateral ratio cannot be less than 100\', code: -32600, method: updateloanscheme')
+  })
+
+  it('should not updateLoanScheme if interestRate is less than 0.01', async () => {
+    const promise = testing.rpc.loan.updateLoanScheme(300, new BigNumber(0.00999), { id: 'scheme1' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\ninterest rate cannot be less than 0.01\', code: -32600, method: updateloanscheme')
+  })
+
+  it('should not updateLoanScheme if same minColRatio and interestRate were created before', async () => {
+    const promise = testing.rpc.loan.updateLoanScheme(100, new BigNumber(1.5), { id: 'scheme1' }) // Failed because its minColRatio and interestRate are same as default
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nLoan scheme default with same interestrate and mincolratio already exists\', code: -32600, method: updateloanscheme')
+  })
+
+  it('should not updateLoanScheme if id does not exist', async () => {
+    const promise = testing.rpc.loan.updateLoanScheme(300, new BigNumber(3.5), { id: 'scheme2' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nCannot find existing loan scheme with id scheme2\', code: -32600, method: updateloanscheme')
+  })
+
+  it('should not updateLoanScheme if id is an empty string', async () => {
+    const promise = testing.rpc.loan.updateLoanScheme(300, new BigNumber(3.5), { id: '' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nid cannot be empty or more than 8 chars long\', code: -32600, method: updateloanscheme')
+  })
+
+  it('should not updateLoanScheme if id is more than 8 chars long', async () => {
+    const promise = testing.rpc.loan.updateLoanScheme(300, new BigNumber(3.5), { id: '123456789' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nid cannot be empty or more than 8 chars long\', code: -32600, method: updateloanscheme')
+  })
+
+  it('should updateLoanScheme at activateAfterBlock which is block 120', async () => {
+    // Wait for block 110
+    await testing.container.waitForBlockHeight(110)
```

Comment:
Seems unnecessary

Reply:
If don't do this, the code for this test unit maybe run in the next it

Reply:
Ok I see